src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -303,7 +305,20 @@ internal async Task<bool> ApplyChangedSettingsAsync(CancellationToken cancellati
             // Unknown exception.
             catch (Exception ex)
             {
-                MessageHelper.ShowErrorMessage(Resources.ShowError_ApplySettingFailed, Resources.ErrorDialogBoxTitle);
+                if (ex is RemoteInvocationException remoteException
```

Comment:
Exceptions are not being catch correctly and it goes directly to this. Should I move all the above exceptions to here? Is this approach correct? 

I found a similar approach in https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/UserInterfaceService/NuGetUI.cs#L379

Reply:
Did you look at [https://github.com/NuGet/NuGet.Client/pull/3711](https://github.com/NuGet/NuGet.Client/pull/3711) from the example that you linked? It seems like you are seeing the same problem.

Reply:
It looks the same, then I should change this function to display the exceptions like this right?

Reply:
I think that would make sense.